week-03/assessment/solar-farm/solar-farm/src/main/java/learn/solar/domain/PanelResult.java

```diff
@@ -0,0 +1,25 @@
+package learn.solar.domain;
+
+import learn.solar.models.Material;
+import learn.solar.models.Panel;
+
+import java.util.ArrayList;
+import java.util.List;
+
+// domain result for handling success/failure
+public class PanelResult {
+    private final ArrayList<String> messages = new ArrayList<>();
+    private Panel panel;
+
+    //messages getter, create a new list
+    public List<String> getMessages(){ return messages;}
+    //Panel getter
+    public Panel getPanel(){return panel;}
+    //Panel setter
+    public void setPanel(Panel panel){ this.panel = panel;}
+    //adds an error message to messages
+    public void addMessage(String message){messages.add(message);}
+    //calculated getter, true if no error messages
+    public boolean isSuccess(){
+        return messages.isEmpty();}
```

Comment:
here you will want to check that the array size is 0 

```suggestion
        return messages.size() == 0; }
```